application/client.core/src/app/environment/components/views/chart/chart/component.ts

```diff
@@ -135,10 +135,17 @@ export class ViewChartCanvasComponent implements AfterViewInit, AfterContentInit
         let position: number | undefined = this._getPositionByChartPointData(event);
         // Try to get data
         if (position === undefined) {
-            const rows: number = this.service.getStreamSize();
+            if (this._ng_charts === undefined) {
+                return;
+            }
+            const visible: boolean = this._isYAxisVisible();
+            const range: IRange = this._getRange();
+            const rows: number = range.end - range.begin;
             const size: ClientRect = (event.target as HTMLElement).getBoundingClientRect();
-            const rate: number = size.width / rows;
-            position = Math.round(event.offsetX / rate);
+            const width: number = size.width - (visible ? this._ng_charts.chartArea.left : 0) ;
```

Comment:
I see, it's from current implementation, but could we move `getBoundingClientRect` out from here as soon as it's quite expensive operation? We also do it with `_resize` method and it would be more optimal if we just store value of `width` (which was gotten with `_resize`) as private property of class and use it here instead addition call of expensive method `getBoundingClientRect`